### PR TITLE
Chore: convert apps/meteor/app/api/server/lib/ files to TS

### PR DESCRIPTION
**Pull request by felipe-rod123**
## Proposed changes (including videos or screenshots)

This pull request converts files on `apps/meteor/app/api/server/lib/` to Typescript.

### CODE REVIEW

File: apps/meteor/app/api/server/lib/rooms.ts [hunk 1]
```diff
@@ -100,10 +117,21 @@ export async function findAdminRoomsAutocomplete({ uid, selector }) {
...
+	selector: { name: string };
+	pagination: { offset: number; count: number; sort: [string, number][] };
+}): Promise<{
+	items: IRoom[];
+	total: number;
+}> {
 	const userRoomsIds = Subscriptions.cachedFindByUserId(uid, { fields: { rid: 1 } })
 		.fetch()
-		.map((item) => item.rid);
+		.map((item: IMessage) => item.rid);
```
Comment (ggazzo) [inline apps/meteor/app/api/server/lib/rooms.ts:134]: `ISubscription `

File: apps/meteor/app/api/server/lib/rooms.ts [hunk 2]
```diff
@@ -146,7 +176,7 @@ export async function findRoomsAvailableForTeams({ uid, name }) {
 
 	const userRooms = Subscriptions.findByUserIdAndRoles(uid, ['owner'], { fields: { rid: 1 } })
 		.fetch()
-		.map((item) => item.rid);
+		.map((item: IMessage) => item.rid);
```
Comment (ggazzo) [inline apps/meteor/app/api/server/lib/rooms.ts:179]:
```suggestion
		.map((item: ISubscription) => item.rid);
```

File: apps/meteor/app/api/server/lib/emoji-custom.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { IEmojiCustom } from '@rocket.chat/core-typings';
+import { SortOptionObject } from 'mongodb';
+
+import { EmojiCustom } from '../../../models/server/raw';
+
+export async function findEmojisCustom({
+	query = {},
+	pagination: { offset, count, sort },
+}: {
+	query: {};
```
Comment (ggazzo) [inline apps/meteor/app/api/server/lib/emoji-custom.ts:10]:
```suggestion
	query: FilterQuery...;
```

**Review by ggazzo — CHANGES_REQUESTED** [review]
please check the suggestions

File: apps/meteor/app/api/server/lib/messages.ts [hunk 4]
```diff
@@ -12,7 +27,7 @@ export async function findMentionedMessages({ uid, roomId, pagination: { offset,
 		throw new Error('invalid-user');
 	}
 
-	const cursor = await Messages.findVisibleByMentionAndRoomId(user.username, roomId, {
+	const cursor = await Messages.findVisibleByMentionAndRoomId((user as IUser).username, roomId, {
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/messages.ts:30]: What if you put this (as IUser) in line 25 instead? The type of `user` there should be IUser | null

File: apps/meteor/app/api/server/lib/rooms.ts [hunk 5]
```diff
@@ -100,10 +117,21 @@ export async function findAdminRoomsAutocomplete({ uid, selector }) {
...
+	selector: { name: string };
+	pagination: { offset: number; count: number; sort: [string, number][] };
+}): Promise<{
+	items: IRoom[];
+	total: number;
+}> {
 	const userRoomsIds = Subscriptions.cachedFindByUserId(uid, { fields: { rid: 1 } })
 		.fetch()
-		.map((item) => item.rid);
+		.map((item: ISubscription) => item.rid);
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/rooms.ts:134]: For consistency, this is not a complete ISubscription object, it's `Pick<ISubscription, 'rid'>`

File: apps/meteor/app/api/server/lib/rooms.ts [hunk 6]
```diff
@@ -66,7 +83,7 @@ export async function findChannelAndPrivateAutocomplete({ uid, selector }) {
 
 	const userRoomsIds = Subscriptions.cachedFindByUserId(uid, { fields: { rid: 1 } })
 		.fetch()
-		.map((item) => item.rid);
+		.map((item: ISubscription) => item.rid);
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/rooms.ts:86]: For consistency, this is not a complete ISubscription object, it's `Pick<ISubscription, 'rid'>`

File: apps/meteor/app/api/server/lib/rooms.ts [hunk 7]
```diff
@@ -146,7 +176,7 @@ export async function findRoomsAvailableForTeams({ uid, name }) {
 
 	const userRooms = Subscriptions.findByUserIdAndRoles(uid, ['owner'], { fields: { rid: 1 } })
 		.fetch()
-		.map((item) => item.rid);
+		.map((item: ISubscription) => item.rid);
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/rooms.ts:179]: For consistency, this is not a complete ISubscription object, it's `Pick<ISubscription, 'rid'>`

File: apps/meteor/app/api/server/lib/users.ts [hunk 8]
```diff
@@ -39,8 +55,8 @@ export async function findUsersToAutocomplete({ uid, selector }) {
  * Returns a new query object with the inclusive fields only
  * @param {Object} query search query for matching rows
  */
-export function getInclusiveFields(query) {
-	const newQuery = {};
+export function getInclusiveFields(query: Query): {} {
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/users.ts:58]: 🤔 this should be something like `{} | { [k; keyof typeof Query]: 1 }`

File: apps/meteor/app/api/server/lib/users.ts [hunk 9]
```diff
@@ -78,7 +103,7 @@ export function getNonEmptyFields(fields) {
  * get the default query if **query** is empty (`{}`) or `undefined`/`null`
  * @param {Object|null|undefined} query the query from parsed jsonQuery
  */
-export function getNonEmptyQuery(query) {
+export function getNonEmptyQuery(query: Query): unknown {
```
Comment (KevLehman) [inline apps/meteor/app/api/server/lib/users.ts:106]: This will return something different from unknown 👀

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**